### PR TITLE
Feat/ls

### LINKED ISSUE #39: LS CompoundMetric
Implement a CompoundMetric for computing linear separability (LS).

Make sure to append to  #44  if applicable.

#35

### PR DESCRIPTION
**Pull request by tediiieee**
Closes #39

### CODE REVIEW

**Review by sporreking — CHANGES_REQUESTED** [review]
fix some tiny things, looks good overall!

File: environment/agent_unetgan.py [hunk 1]
```diff
@@ -197,7 +197,6 @@ def sample_generator(G, config: dict, z: torch.Tensor) -> None:
     z_batches = torch.split(z, config["batch_size"])
     n_batches = len(z_batches)
 
-    print(f"Generating {n} images in {n_batches} batches.")
```
[the remark this answers is not in the document]
Comment (tediiieee) [reply environment/agent_unetgan.py:200]: Used tqdm instead

File: src/population/Population.py [hunk 2]
```diff
@@ -28,13 +28,25 @@ def __init__(self, name: str):
...
+        self._create_metadata_folder()
+
         self._data = self._file_jar.get_file(
             Population.POPULATION_DATA_FILE_NAME, pd.read_pickle
         )
 
+    def _create_metadata_folder(self) -> None:
+        # Check that metadata_dir is a directory.
+        if not self._metadata_dir.is_dir():
+            if not self._metadata_dir.exists():
```
[the remark this answers is not in the document]
Comment (tediiieee) [reply src/population/Population.py:43]: Same check as in init, i can change both

[the remark this answers is not in the document]
Comment (tediiieee) [reply src/population/Population.py:46]: added check for file

File: src/population/Population.py [hunk 3]
```diff
@@ -26,15 +26,34 @@ def __init__(self, name: str):
...
         )
 
+    def _create_metadata_directory(self) -> None:
+        # Check that metadata_dir is a directory.
+        if not self._metadata_dir.is_dir():
+            if not self._metadata_dir.exists():
+                self._metadata_dir.mkdir(parents=True)
+            elif self._metadata_dir.is_file():
+                raise FileNotFoundError(
+                    "File named 'metadata' is not allowed in the population folder."
```
Comment (sporreking) [inline src/population/Population.py:50]: ctrl+F "folder" -> "directory"

**Review by sporreking — APPROVED** [review]
WP

**Pull request merged**